src/org/zaproxy/zap/extension/pscanrulesAlpha/XChromeLoggerDataInfoLeakScanner.java

```diff
@@ -0,0 +1,133 @@
+
+/**
+ * X-ChromeLogger-Data header information leak passive scan rule 
+ * @author kingthorin+owaspzap@gmail.com
+ */
+public class XChromeLoggerDataInfoLeakScanner extends PluginPassiveScanner{
+
+	private static final String MESSAGE_PREFIX = "pscanalpha.xchromeloggerdata.";
+	private static final int PLUGIN_ID = 10052;
+	
+	private PassiveScanThread parent = null;
+	private static final Logger logger = Logger.getLogger(XChromeLoggerDataInfoLeakScanner.class);
+	
+	@Override
+	public void setParent(PassiveScanThread parent) {
+		this.parent = parent;
+	}
+
+	@Override
+	public void scanHttpRequestSend(HttpMessage msg, int id) {
+		// Only checking the response for this plugin
+	}
+	
+	@Override
+	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		long start = System.currentTimeMillis();
+
+		Vector<String> xcldHeader = msg.getResponseHeader().getHeaders("X-ChromeLogger-Data"); //Get the header(s)
+		Vector<String> xcpdHeader = msg.getResponseHeader().getHeaders("X-ChromePhp-Data"); //Add any header(s) using the alternate name
+		
+		List<String> loggerHeaders = new ArrayList<String>(2);
+		
+		if(xcldHeader != null && !xcldHeader.isEmpty()) {
+			loggerHeaders.addAll(xcldHeader);
+		}
+		if(xcpdHeader !=null && !xcpdHeader.isEmpty()) {
+			loggerHeaders.addAll(xcpdHeader);
+		}
+
+		if (!loggerHeaders.isEmpty()) { //Header(s) Found
+			for (String xcldField : loggerHeaders) {
+					Alert alert = new Alert(getPluginId(), Alert.RISK_MEDIUM, Alert.CONFIDENCE_HIGH, //PluginID, Risk, Reliability
+						getName()); 
+		    			alert.setDetail(
+		    					getDescription(), //Description
+		    					msg.getRequestHeader().getURI().toString(), //URI
+		    					"",	// Param
+		    					"", // Attack
+		    					getOtherInfo(xcldField), // Other info
+		    					getSolution(), //Solution
+		    					getReference(), //References
+		    					xcldField,	// Evidence 
+		    					200, // CWE Id 
+		    					13,	// WASC Id 
+		    					msg); //HttpMessage
+		    		parent.raiseAlert(id, alert);
+				}
+			}
+	    	if (logger.isDebugEnabled()) {
+	    		logger.debug("\tScan of record " + id + " took " + (System.currentTimeMillis() - start) + " ms");
+	    	}
+	}
+
+	@Override
+	public int getPluginId() {
+		return PLUGIN_ID;
+	}
+	
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+	
+	private String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	private String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	private String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+	private String getOtherInfo(String headerValue) {
+		try {
+    		byte[] decodedByteArray = Base64.decode(headerValue);
+            return Constant.messages.getString(MESSAGE_PREFIX + "otherinfo.msg") + "\n" + new String(decodedByteArray);

```

Comment:
Consider using the constructor `String(byte[], Charset)`, with the second parameter as `StandardCharsets.UTF_8`.
The `String(byte[])` constructor behaviour depends on the platform's Charset (which might yield different results).